recipes/ignition-math/all/conanfile.py

```diff
@@ -68,7 +68,10 @@ def requirements(self):
         self.requires("eigen/3.3.9")
 
     def build_requirements(self):
-        self.build_requires("ignition-cmake/2.5.0")
+        if int(self.version.split(".")[1]) <= 8:
```

Comment:
```suggestion
        if tools.Version(self.version).minor <= "8":
```

Reply:
@uilianries  this fix doesn't seem to work when ignition-math minor versions are greater than 9, since python comparison of string is in lexicographic order, meaning  ` "9" >= "8" ` results in `True` but `"10" >= "8"` results in `False` since first the characer `"1"` is compared against character `"8"` and it results in `False` therefore the comparison `"1" >= "8"` stops early and returns `False` for the entire comparison. Conversely `"10" < "8"` results in `True`  

Here's an screenshot of the comparison on my console
![image](https://user-images.githubusercontent.com/9218964/158167760-36a11c00-bd35-4e6a-9145-8dc1e030dcde.png)

I'm reverting this change back unless you have a better suggestion

Reply:
Thank you for your check, please, see my updated suggestion.

Reply:
I did exactly [this](https://github.com/conan-io/conan-center-index/pull/9569/files#diff-68a42dbaa991102b8ab0fa45fa4ed20dae5b48e127822865644e071974f6be07R71)  :wink: